third_party/asr/asr582x/asr_sdk.gni

```diff
@@ -0,0 +1,126 @@
+  }
+
+  # While most asr sdk includes come from the asr_sdk_config config (in
+  # BUILD.gn), this template gets instantiated by each project each of which
+  # provides a few extra includes/defines. Handle those here.
+  config("project_configs") {
+    include_dirs = []
+    if (defined(invoker.include_dirs)) {
+      include_dirs += invoker.include_dirs
+    }
+    include_dirs += [ "${asr_project_dir}/include" ]
+
+    defines = []
+    if (defined(invoker.defines)) {
+      defines += invoker.defines
+    }
+  }
+
+  sdk_target_name = target_name
+
+  # Add sources here
+  source_set(sdk_target_name) {
+    sources = []
+    if (defined(invoker.sources)) {
+      sources += invoker.sources
+    }
+
+    _third_party_src = [
+#at
+        "at_cmd/atcmd_comm.c",
+        "at_cmd/atcmd_user.c",
+        "at_cmd/atcmd_matter.c",
+        "peripheral/duet/src/duet_boot.c",
+        "peripheral/duet/src/duet_board.c",
+        "peripheral/duet/src/duet_efuse.c",
+        "peripheral/duet/src/duet_flash.c",
+        "peripheral/duet/src/duet_flash_alg.c",
+        "peripheral/duet/src/duet_wdg.c",
+        "peripheral/duet/src/duet_timer.c",
+        "peripheral/duet/src/duet_gpio.c",
+        "peripheral/duet/src/duet_i2c.c",
+        "peripheral/duet/src/duet_rf_spi.c",
+        "peripheral/duet/src/duet_pwm.c",
+        "peripheral/duet/src/duet_rtc.c",
+        "peripheral/duet/src/duet_pinmux.c",
+        "peripheral/duet/src/duet_uart.c",
+        "peripheral/duet/src/duet_adc.c",
+        "peripheral/duet/src/duet_dma.c",
+        "platform/duet/system/sysCalls.c",
+        "platform/duet/system/soc_init.c",
+        "platform/duet/system/printf_uart.c",
+        "platform/duet/system/printf-stdarg.c",
+        "platform/duet/system/system_cm4.c",
+        "platform/duet/system/startup_cm4.S",
+        "platform/duet/ethernetif_wifi.c",
+#rtos
+        "platform/duet/lega_rtos.c",
+        "freertos/Source/croutine.c",
+        "freertos/Source/event_groups.c",
+        "freertos/Source/list.c",
+        "freertos/Source/queue.c",
+        "freertos/Source/tasks.c",
+        "freertos/Source/timers.c",
+        "freertos/Source/portable/MemMang/heap_5.c",
+        "freertos/Source/portable/GCC/ARM_CM4F/port.c",
+#ble
+        "demo/matter_demo/app/ble_demo/duet_hl_matter/app.c",
+        "demo/matter_demo/app/ble_demo/duet_hl_matter/user_platform.c",
+        "matter/asr_factory_config.c",
+#elog
+        "common/duet/easylogger/elog_async.c",
+        "common/duet/easylogger/elog_buf.c",
+        "common/duet/easylogger/elog_module.c",
+        "common/duet/easylogger/elog_port.c",
+        "common/duet/easylogger/elog_utils.c",
+        "common/duet/easylogger/elog.c",
+    ]
+
+    foreach(src, _third_party_src) {
+      sources += [ rebase_path(src,
+                               "${asr_project_dir}",
+                               "${asr_sdk_build_root}/asr_sdk/") ]
+    }
+
+    public_deps = []
```

Comment:
This is initialized with empty and also unconditionally appended below. Start it off non-empty instead to reduce line count and complexity. Ordering in deps should not matter